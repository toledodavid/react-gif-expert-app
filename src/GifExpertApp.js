import React, { useState } from 'react';
import AddCategory from './components/AddCategory';

const GifExpertApp = () => {

  const [categories, setCategories] = useState(['One punch', 'Samurai X', 'Dragon Ball']);


  // const handleAdd = () => {
  //   // setCategories([...categories, 'Daredevil']);
  //   setCategories(categories => [...categories, 'Batman']);
  // }

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr/>

      <ol>
        {
          categories.map(category => {
          return <li key={category}>{category}</li>
          })
        }
      </ol>
    </>
  );
}

export default GifExpertApp;