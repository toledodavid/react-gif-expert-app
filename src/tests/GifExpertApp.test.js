import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Tests for <GifExpertApp /> component', () => {

  test('It should render <GifExpertApp /> correctly', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('It should show a list of categories', () => {
    const categories = ['Batman', 'Superman'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });

});