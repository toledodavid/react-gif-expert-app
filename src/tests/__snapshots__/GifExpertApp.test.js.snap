// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifExpertApp /> component It should render <GifExpertApp /> correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Tests for <GifExpertApp /> component It should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Batman"
      key="Batman"
    />
    <GifGrid
      category="Superman"
      key="Superman"
    />
  </ol>
</Fragment>
`;
